data/src/main/java/com/linkedin/data/ByteString.java

```diff
@@ -481,6 +481,26 @@ public int feed(ByteArrayFeeder feeder, int index) throws IOException
     return returnIndex < _byteArrays.getArraySize() ? returnIndex : -1;
   }
 
+  /**
+   * Feeds a chunk of this {@link ByteString} to a @{@link com.linkedin.data.Data.DataParser}
+   * without copying the underlying byte[].
+   *
+   * @param parser the feeder to feed the bytes to
+   * @param index the index of the chunk to feed
+   *
+   * @throws IOException if an error occurs while writing to the feeder
+   *
+   * @return The next index to feed or -1 if no more indices are left to feed.
+   */
+  public int feed(Data.DataParser parser, int index) throws IOException
```

Comment:
Would it be possible to reuse the implementation of `#feed(ByteArrayFeeder, int)` by having `Data.DataParser` implement `ByteArrayFeeder`? It looks like it already satisfies that interface anyway.

Reply:
ByteArrayFeeder is part of jackson lib, using it for generic parser here would be weird, that's why copied required interface methods for us here.

Reply:
Ah, I see. Yeah we would have less control over the interface then, could cause difficulties.